xarray/backends/common.py

```diff
@@ -27,8 +22,41 @@
 NONE_VAR_NAME = '__values__'
 
 
-# dask.utils.SerializableLock if available, otherwise just a threading.Lock
-GLOBAL_LOCK = Lock()
+def get_scheduler(get=None, collection=None):
+    try:
+        from dask.utils import effective_get
+        actual_get = effective_get(get, collection)
+        try:
+            from dask.distributed import Client
+            if isinstance(actual_get.__self__, Client):
+                return 'distributed'
+        except (ImportError, AttributeError):
+            import dask.multiprocessing
+            if actual_get == dask.multiprocessing.get:
+                return 'multiprocessing'
+            else:
+                return 'threaded'
+    except ImportError:
+        return None
+
+
+def get_scheduler_lock(scheduler):
+    if scheduler == 'distributed':
+        from dask.distributed import Lock
+        return Lock
+    elif scheduler == 'multiprocessing':
+        import multiprocessing as mp
+        return mp.Manager().Lock
+    elif scheduler == 'threaded':
+        from dask.utils import SerializableLock
+        return SerializableLock
+    else:
+        from threading import Lock
+        return Lock
+
+
+SCHEDULER = get_scheduler()
+GLOBAL_LOCK = get_scheduler_lock(SCHEDULER)()
```

Comment:
Actually, for what we need here (we should call it `HDF5_LOCK` for clarity), we probably don't even need something to be scheduler specific at all. It can be a SerializableLock if dask is installed, or otherwise just a threading.Lock.